api/src/orgs/data.js

```diff
@@ -39,6 +46,55 @@ export function getOrganisations (
   ).then(withOrgStats(query, { ...filter }))
 }
 
+export async function updateProfile (
+  db,
+  args
+) {
+  const { address, signerAddress, signedMessage, ...updateParams } = args
+
+  const originalMessage = composeSignedMessage(address, updateParams)
+  const isAddressValid = validateSignerAddress(originalMessage, signedMessage, signerAddress)
+  Web3EthContract.setProvider('wss://mainnet.eth.aragon.network/ws')
+
+  const kernelContract = new Web3EthContract(kernelAbi, address)
+
+  try {
+    const hasPermission = await kernelContract.methods.hasPermission(
+      signerAddress,
+      address,
+      MANAGE_PROFILE_ROLE,
+      EMPTY_SCRIPT
+    ).call()
+    if (!isAddressValid || !hasPermission) {
+      throw new Error('Failed message verification.')
+    }
+  } catch (err) {
+    console.error(err)
+    return
+  }
+
+  const { profile = {} } = await db.collection('orgs').findOne({ address })
+  const currentEditors = profile.editors || []
+  const newEditors = [...currentEditors, signerAddress]
+  // Remove duplicates
+  const filteredEditors = new Set(newEditors)
+  const newFilteredEditors = [...filteredEditors]
+
+  await db.collection('orgs').updateOne(
+    { address },
+    { $set: { profile: { ...profile, ...updateParams, editors: newFilteredEditors } } })
+
+  return db.collection('orgs').findOne({ address })
+}
+
+export async function getSingleOrganisation (
```

Comment:
I think just `getOrganisation` is fine. Also there is no need for this function to be `async` since it returns a promise and does not use `await`

Reply:
Done in https://github.com/1Hive/apiary/pull/181/commits/47385dfa29ed128f82c09f3045de9cbefc67de6f.